sound.js

```diff
@@ -2,18 +2,24 @@
 
 var RNSound = require('react-native').NativeModules.RNSound;
 var IsAndroid = RNSound.IsAndroid;
-
+var resolveAssetSource = require("react-native/Libraries/Image/resolveAssetSource");
 var nextKey = 0;
 
 function isRelativePath(path) {
   return !/^\//.test(path);
 }
 
 function Sound(filename, basePath, onError) {
-  this._filename = basePath ? basePath + '/' + filename : filename;
-
-  if (IsAndroid && !basePath && isRelativePath(filename)) {
-    this._filename = filename.toLowerCase().replace(/\.[^.]+$/, '');
+  var image = resolveAssetSource(filename);
```

Comment:
Yeah good point. I totally spaced when writing this. I just renamed it to `asset` instead.